### PR TITLE
dnsproxy: 0.54.0 -> 0.65.2

### PR DESCRIPTION
**Pull request by diogotcorreia**
## Description of changes

https://github.com/AdguardTeam/dnsproxy/releases

~~While 0.65.0 has already been released, it is not possible to update to that version at the moment in this branch since it requires Go 1.21.7, which is still in the staging branch: https://github.com/NixOS/nixpkgs/pull/286825~~

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by contrun** [comment]
Thank you for the contribution. Unfortunately, I can't build this with `nixpkgs-review pr 287998`, whose output is

```
no oauth token
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/287998/head:refs/nixpkgs-review/1
remote: Enumerating objects: 14, done.
remote: Counting objects: 100% (8/8), done.
remote: Total 14 (delta 8), reused 8 (delta 8), pack-reused 6
Unpacking objects: 100% (14/14), 3.15 KiB | 461.00 KiB/s, done.
From https://github.com/NixOS/nixpkgs
 + 1c7ec861d1de...76f0737593d0 refs/pull/287998/head -> refs/nixpkgs-review/1  (forced update)
$ git worktree add /home/vagrant/.cache/nixpkgs-review/pr-287998/nixpkgs 8b5828213f4e22e24d8e461fe24513e5b86564b7
Preparing worktree (detached HEAD 8b5828213f4e)
Updating files: 100% (39630/39630), done.
HEAD is now at 8b5828213f4e Merge pull request #288982 from fabaff/aiohue-bump
$ git merge --no-commit --no-ff 76f0737593d02f0506111b421751d9d4c343a9d3
Auto-merging pkgs/tools/networking/dnsproxy/default.nix
Automatic merge went well; stopped before committing as requested
error:
       … while calling the 'listToAttrs' builtin

         at /nix/store/rmc9pzj3fi8nwd93qvgy6s09jnvnx170-nixpkgs-review-2.10.3/lib/python3.11/site-packages/nixpkgs_review/nix/evalAttrs.nix:46:1:

           45|
           46| listToAttrs (concatMap getProperties attrs)
             | ^
           47|

       … while calling the 'concatMap' builtin

         at /nix/store/rmc9pzj3fi8nwd93qvgy6s09jnvnx170-nixpkgs-review-2.10.3/lib/python3.11/site-packages/nixpkgs_review/nix/evalAttrs.nix:46:14:

           45|
           46| listToAttrs (concatMap getProperties attrs)
             |              ^
           47|

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: attribute 'mainProgram' already defined at /home/vagrant/.cache/nixpkgs-review/pr-287998/nixpkgs/pkgs/tools/networking/dnsproxy/default.nix:27:5

       at /home/vagrant/.cache/nixpkgs-review/pr-287998/nixpkgs/pkgs/tools/networking/dnsproxy/default.nix:29:5:

           28|     maintainers = with maintainers; [ contrun ];
           29|     mainProgram = "dnsproxy";
             |     ^
           30|   };
https://github.com/NixOS/nixpkgs/pull/287998 failed to build: nix --extra-experimental-features nix-command no-url-literals --system x86_64-linux eval --nix-path nixpkgs=/home/vagrant/.cache/nixpkgs-review/pr-287998/nixpkgs nixpkgs-overlays=/tmp/tmpdkm5_azv --json --impure --no-allow-import-from-derivation --expr (import /nix/store/rmc9pzj3fi8nwd93qvgy6s09jnvnx170-nixpkgs-review-2.10.3/lib/python3.11/site-packages/nixpkgs_review/nix/evalAttrs.nix { attr-json = /tmp/tmp0hed8jis; }) failed to run, /tmp/tmp0hed8jis was stored inspection
$ git worktree remove -f /home/vagrant/.cache/nixpkgs-review/pr-287998/nixpkgs

```

There is also https://github.com/NixOS/nixpkgs/pull/288686 which tried to update dnsproxy to 0.65.0

**Comment by diogotcorreia** [comment]
@contrun It seems like commit https://github.com/NixOS/nixpkgs/commit/8121f3559a98259a8e767dedf4eaf3939442c54d added `mainProgram` as well.
I'll rebase the PR in a few minutes, it should work after that.

The PR you mention tries to upgrade to a version that requires Go 1.27.7, which is still in staging iirc.

**Comment by contrun** [comment]
in that case, I think we can merge this PR first.

**Comment by diogotcorreia** [comment]
Should be good now
I'll run `nixpkgs-review` as well

**Comment by diogotcorreia** [comment]
Seems to have worked:

```
❯ nixpkgs-review pr 287998
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/287998/head:refs/nixpkgs-review/1
From https://github.com/NixOS/nixpkgs
 * [new branch]                master                -> refs/nixpkgs-review/0
 * [new ref]                   refs/pull/287998/head -> refs/nixpkgs-review/1
$ git worktree add /home/dtc/.cache/nixpkgs-review/pr-287998/nixpkgs fcc8a48e119129c37af24036986957b47b743447
Preparing worktree (detached HEAD fcc8a48e119)
remote: Enumerating objects: 46, done.
remote: Counting objects: 100% (41/41), done.
remote: Compressing objects: 100% (39/39), done.
remote: Total 46 (delta 4), reused 24 (delta 2), pack-reused 5
Receiving objects: 100% (46/46), 622.12 KiB | 2.88 MiB/s, done.
Resolving deltas: 100% (4/4), done.
Updating files: 100% (39630/39630), done.
HEAD is now at fcc8a48e1191 Merge pull request #290291 from anmonteiro/anmonteiro/fix-logs-override
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/dtc/.cache/nixpkgs-review/pr-287998/nixpkgs nixpkgs-overlays=/tmp/tmpp9gp0auf -qaP --xml --out-path --show-trace --no-allow-import-from-derivation
$ git merge --no-commit --no-ff 51c53b4aa2fba27d49aac33dfe73503da96483a8
Automatic merge went well; stopped before committing as requested
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/dtc/.cache/nixpkgs-review/pr-287998/nixpkgs nixpkgs-overlays=/tmp/tmpp9gp0auf -qaP --xml --out-path --show-trace --no-allow-import-from-derivation --meta
1 package updated:
dnsproxy (0.54.0 → 0.64.1)

$ nix build --nix-path nixpkgs=/home/dtc/.cache/nixpkgs-review/pr-287998/nixpkgs nixpkgs-overlays=/tmp/tmpp9gp0auf --extra-experimental-features nix-command no-url-literals --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed -f /home/dtc/.cache/nixpkgs-review/pr-287998/build.nix

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/287998

1 package built:
dnsproxy

$ /nix/store/j7nl2pj606d8ld5818hw3z3fbz00sdc5-nix-2.18.1/bin/nix-shell /home/dtc/.cache/nixpkgs-review/pr-287998/shell.nix --nix-path nixpkgs=/home/dtc/.cache/nixpkgs-review/pr-287998/nixpkgs nixpkgs-overlays=/tmp/tmpp9gp0auf
```

PS: If you want to, I don't mind being a co-maintainer of this package.

**Comment by contrun** [comment]
That's awesome. Please do add yourself to the maintenance.

**Review by contrun — APPROVED** [review]
It's now working. Thanks.

File: pkgs/tools/networking/dnsproxy/default.nix [hunk 1]
```diff
@@ -2,26 +2,29 @@
 
 buildGoModule rec {
   pname = "dnsproxy";
-  version = "0.54.0";
+  version = "0.64.1";
 
   src = fetchFromGitHub {
     owner = "AdguardTeam";
     repo = pname;
```
Comment (dotlambda) [inline pkgs/tools/networking/dnsproxy/default.nix:9]:
```suggestion
    repo = "dnsproxy";
```

File: pkgs/tools/networking/dnsproxy/default.nix [hunk 2]
```diff
@@ -2,26 +2,29 @@
...
     rev = "v${version}";
-    sha256 = "sha256-wA88v8zF5sq3NQwKh8g4k/COviuaegGn6bBTzBMcdGM=";
+    sha256 = "sha256-w5eGMf6noQh8K2TlHmT7Qj/pU96J3Y1qrn4inrQ2FL4=";
```
Comment (dotlambda) [inline pkgs/tools/networking/dnsproxy/default.nix:11]:
```suggestion
    hash = "sha256-w5eGMf6noQh8K2TlHmT7Qj/pU96J3Y1qrn4inrQ2FL4=";
```

**Review by contrun — APPROVED** [review]

**Comment by diogotcorreia** [comment]
Since Go 1.21.7 is now in the master branch, I've updated to dnsproxy 0.65.2.
This is ready to merge :)

**Pull request merged**